pkg/infra/db.go

```diff
@@ -21,6 +21,12 @@ const (
 	dbMaxOpenConns = 5
 )
 
+type DBer interface {
+	ExecContext(ctx context.Context, query string, args ...any) (sql.Result, error)
+	QueryContext(ctx context.Context, query string, args ...any) (*sql.Rows, error)
+	QueryRowContext(ctx context.Context, query string, args ...any) *sql.Row
+}
+
```

Comment:
```go
type DBContextExecQuerier interface {}
```

```go
type DBExecQuerier interface {}
```

both seem a bit weird because you might expect that ExecQuerier is made up of the DBExecuter and DBQuerier but their interfaces are different.. i.e.

```go
type DBExecQuerier interface {
  DBExecutor
  DBQuerier
}
```

Reply:
Hmm....the `-er` format seems to fit better for (/ is documented as the pattern for https://go.dev/doc/effective_go#interface-names) one-method or similar-method interfaces. I wonder if the `DB()` func should be more like `InitDB()` or `CreateDB` and this interface should just be `DB`, more akin to how an HTTP request client interface would look?